### PR TITLE
Moving `detect_nuclei_kofahi` method to histomicstk

### PR DESCRIPTION
**Pull request by fbudin69500**
`detect_nuclei_kofahi` was implemented on the server-side. It was moved directly
into histomicstk, as a new function, to facilitate access to the algorithm.

### CODE REVIEW

**Comment by fbudin69500** [comment]
@cdeepakroy Moving kofahi to histomicstk.

File: histomicstk/segmentation/nuclear/detect_nuclei_kofahi.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    im_nuclei_stain : array_like
+        An intensity image used  to extract the image foreground using a
+        graph-cuts-based binarization and to detect nuclear seed points which
+         are used to perform an initial segmentation that is refined using a
+        second graph-cuts-based algorithm. Dimensions M x N.
+    foreground_threshold : float
+    min_radius : float
+    max_radius : float
+    min_nucleus_area : int
+    local_max_search_radius : float
```
Comment (cdeepakroy) [inline histomicstk/segmentation/nuclear/detect_nuclei_kofahi.py]: Can you add a description to all these parameters? You can find them either in the NucleiDetection XML or within other functions it calls withing this function.

File: histomicstk/segmentation/nuclear/detect_nuclei_kofahi.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import scipy as sp
+import skimage.morphology
+import histomicstk.filters.shape as htk_shape_filters
+import histomicstk as htk
+
+
+def detect_nuclei_kofahi(im_nuclei_stain, foreground_threshold, min_radius,
+                         max_radius, min_nucleus_area, local_max_search_radius):
+
+    """Performs a nuclear segmentation using kofahi's method.
```
Comment (cdeepakroy) [inline histomicstk/segmentation/nuclear/detect_nuclei_kofahi.py]: Add a brief description of the method here. In short, it uses scale-adaptive multi-scale Laplacian-of-Gaussian filtering for blob enhancement + local maximum clustering for segmentation. The original paper uses Laplacian of Gaussian but we replace it Difference of Gaussian to improve speed.

File: histomicstk/segmentation/nuclear/detect_nuclei_kofahi.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    Parameters
+    ----------
+    im_nuclei_stain : array_like
+        An intensity image used  to extract the image foreground using a
+        graph-cuts-based binarization and to detect nuclear seed points which
+         are used to perform an initial segmentation that is refined using a
+        second graph-cuts-based algorithm. Dimensions M x N.
```
Comment (cdeepakroy) [inline histomicstk/segmentation/nuclear/detect_nuclei_kofahi.py]: This description is incorrect. It uses a nuclei stain (usually hematoxylin) image typically obtained after doing color deconvolution.

**Comment by fbudin69500** [comment]
@cdeepakroy The tests pass now that you clean the cache. Thanks!

**Comment by fbudin69500** [comment]
@cdeepakroy If you think this PR is ready to be merged, can you approve it?

File: histomicstk/segmentation/nuclear/detect_nuclei_kofahi.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import scipy as sp
+import skimage.morphology
+import histomicstk.filters.shape as htk_shape_filters
+import histomicstk as htk
+
+
+def detect_nuclei_kofahi(im_nuclei_stain, foreground_threshold, min_radius,
```
Comment (cdeepakroy) [inline histomicstk/segmentation/nuclear/detect_nuclei_kofahi.py]: change the name of this function to `kofahi_alog_lmc` to summarize the key idea of the method in the func name.

**Pull request merged**